compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -60,27 +59,7 @@ pub struct CompileTimeInterpreter<'mir, 'tcx> {
     pub(super) can_access_statics: CanAccessStatics,
 
     /// Whether to check alignment during evaluation.
-    pub(super) check_alignment: CheckAlignment,
-}
-
-#[derive(Copy, Clone)]
-pub enum CheckAlignment {
-    /// Ignore alignment when following relocations.
-    /// This is mainly used in interning.
-    No,
-    /// Hard error when dereferencing a misaligned pointer.
-    Error,
-    /// Emit a future incompat lint when dereferencing a misaligned pointer.
-    FutureIncompat,
```

Comment:
we could keep the enum and just remove this variant. That would avoid adding `/* CheckAlignment */` comments

Reply:
Fair, done.